pkg/ingestr/operator.go

```diff
@@ -0,0 +1,74 @@
+package ingestr
+
+import (
+	"context"
+	"errors"
+
+	"github.com/bruin-data/bruin/pkg/executor"
+	"github.com/bruin-data/bruin/pkg/pipeline"
+	"github.com/bruin-data/bruin/pkg/scheduler"
+)
+
+type BasicOperator struct{}
+
+func NewBasicOperator() *BasicOperator {
+	return &BasicOperator{}
+}
+
+func (o BasicOperator) Run(ctx context.Context, ti scheduler.TaskInstance) error {
+	return o.RunTask(ctx, ti.GetPipeline(), ti.GetAsset())
+}
+
+func (o BasicOperator) RunTask(ctx context.Context, p *pipeline.Pipeline, t *pipeline.Asset) error {
+	return errors.New("Ingestr tasks not implemented")
+}
+
+type IngestrCheckOperator struct {
+	configs *map[pipeline.AssetType]executor.Config
+}
+
+func (i IngestrCheckOperator) Run(ctx context.Context, ti scheduler.TaskInstance) error {
+	asset := ti.GetAsset()
+	assetType, err := asset.GetIngestrDestinationType()
```

Comment:
I think this is very specific to the operator itself, I wouldn't put this method on the asset itself

Reply:
Thing is that it's needed not only in the operator but also outside so I thought it was a nice abstraction. Will create a function function in the `ingestr` package instead like

```go
func GetIngestrDestinationType(asset Asset) (AssetType, error)
```